presto-geospatial/src/main/java/com/facebook/presto/plugin/geospatial/GeoFunctions.java

```diff
@@ -228,6 +231,32 @@ public static Boolean stIsEmpty(@SqlType(GEOMETRY_TYPE_NAME) Slice input)
         return deserialize(input).isEmpty();
     }
 
+    @Description("Returns true if the input geometry is well formed")
+    @ScalarFunction("ST_IsValid")
+    @SqlType(StandardTypes.BOOLEAN)
+    public static boolean stIsValid(@SqlType(GEOMETRY_TYPE_NAME) Slice input)
+    {
+        return GeometryJtsUtils.deserialize(input).isValid();
+    }
+
+    @Description("If input geometry is not valid, return a reason why. Otherwise, returns null.")
+    @ScalarFunction("geometry_is_valid_reason")
+    @SqlType(StandardTypes.VARCHAR)
+    @SqlNullable
+    public static Slice isValidReason(@SqlType(GEOMETRY_TYPE_NAME) Slice input)
+    {
+        if (input == null) {
+            return null;
+        }
+        Geometry geometry = GeometryJtsUtils.deserialize(input);
```

Comment:
`geometry` can be null (deserialize can return null).

Reply:
`deserialize` returns null only for null input, but I'll move the check for null after call to deserialize for clarity.